cdap/resource_application.go

```diff
@@ -47,49 +42,22 @@ func resourceApplication() *schema.Resource {
 					return defaultNamespace, nil
 				},
 			},
-			"description": {
+			"name": {
 				Type:        schema.TypeString,
-				Optional:    true,
-				ForceNew:    true,
-				Description: "A user friendly description of the application.",
-			},
-			"artifact": {
-				Type:        schema.TypeList,
+				Description: "The name of the application. This will be used as the unique identifier in the CDAP API.",
 				Required:    true,
 				ForceNew:    true,
-				Description: "The artifact used to create the pipeline",
-				MaxItems:    1,
-				Elem: &schema.Resource{
-					Schema: map[string]*schema.Schema{
-						"name": {
-							Type:        schema.TypeString,
-							Required:    true,
-							ForceNew:    true,
-							Description: "The name of the artifact.",
-						},
-						"version": {
-							Type:        schema.TypeString,
-							Required:    true,
-							ForceNew:    true,
-							Description: "The version of the artifact.",
-						},
-						"scope": {
-							Type:        schema.TypeString,
-							Optional:    true,
-							ForceNew:    true,
-							Description: "The scope of the artifact, one of either SYSTEM or USER. Defaults to SYSTEM.",
-							DefaultFunc: func() (interface{}, error) {
-								return "SYSTEM", nil
-							},
-						},
-					},
-				},
 			},
-			"config": {
+			"spec": {
 				Type:        schema.TypeString,
+				Description: "The full contents of the exported pipeline JSON spec.",
 				Required:    true,
 				ForceNew:    true,
-				Description: "The JSON encoded configuration of the pipeline",
+                ValidateFunc: validation.ValidateJsonString,
```

Comment:
Might need to run go fmt to fix indentation

Reply:
done.